formulas/mo/files/mo

```diff
@@ -19,28 +27,80 @@
 
 # Public: Template parser function.  Writes templates to stdout.
 #
-# $0 - Name of the mo file, used for getting the help message.
-# $* - Filenames to parse.  Can use -h or --help as the only option
-#      in order to show a help message.
+# $0            - Name of the mo file, used for getting the help message.
+# --false       - Treat "false" as an empty value.  You may set the
+#                 MO_FALSE_IS_EMPTY environment variable instead to a non-empty
+#                 value to enable this behavior.
+# --help        - Display a help message.
+# --source=FILE - Source a file into the environment before processint
+#                 template files.
+# --            - Used to indicate the end of options.  You may optionally
+#                 use this when filenames may start with two hyphens.
+# $@            - Filenames to parse.
+#
+# Mo uses the following environment variables:
+#
+# MO_FALSE_IS_EMPTY   - When set to a non-empty value, the string "false"
+#                       will be treated as an empty value for the purposes
+#                       of conditionals.
+# MO_ORIGINAL_COMMAND - Used to find the `mo` program in order to generate
+#                       a help message.
 #
 # Returns nothing.
 mo() (
     # This function executes in a subshell so IFS is reset.
     # Namespace this variable so we don't conflict with desired values.
-    local moContent
+    local moContent f2source files doubleHyphens
 
     IFS=$' \n\t'
-    
+    files=()
+    doubleHyphens=false
+
     if [[ $# -gt 0 ]]; then
-        case "$1" in
-            -h|--h|--he|--hel|--help)
-                moUsage "$0"
-                exit 0
-                ;;
-        esac
+        for arg in "$@"; do
+            if $doubleHyphens; then
+                # After we encounter two hyphens together, all the rest
+                # of the arguments are files.
+                files=("${files[@]}" "$arg")
+            else
+                case "$arg" in
+                    -h|--h|--he|--hel|--help|-\?)
+                        moUsage "$0"
+                        exit 0
+                        ;;
+
+                    --false)
+                        # shellcheck disable=SC2030
+                        MO_FALSE_IS_EMPTY=true
+                        ;;
+
+                    --source=*)
+                        f2source="${1#--source=}"
+
+                        if [[ -f "$f2source" ]]; then
+                            # shellcheck disable=SC1090
+                            . "$f2source"
+                        else
+                            echo "No such file: $f2source" >&2
+                            exit 1
+                        fi
+                        ;;
+
+                    --)
+                        # Set a flag indicating we've encountered double hyphens
+                        doubleHyphens=true
+                        ;;
+
```

Comment:
That makes sense, the ? was meant as any single character rather then actual ?. If it doesn't error on not finding the file then no harm done anyways.